harness/determined/searcher/search_method.py

```diff
@@ -182,18 +206,42 @@ def on_trial_exited_early(
         """
         pass
 
-    def save_checkpoint(self, event_id: int) -> None:
+    def save(self, path: pathlib.Path, *, experiment_id: int, event_id: int) -> None:
         """
-        This is optionally implemented to save a checkpoint indexed by event id.
+        This is optionally overridden to save a checkpoint indexed by event id.
         It will be called by the ``SearchRunner`` after receiving operations
         from the ``SearchMethod``
         """
+        searcher_state_file = path.joinpath(STATE_FILE)
+        self.save_method_state(path, event_id)
+        d = self.searcher_state.to_dict()
+        d["experimentId"] = experiment_id
+        with searcher_state_file.open("w") as f:
+            json.dump(d, f)
+
+    def save_method_state(self, path: pathlib.Path, event_id: int) -> None:
```

Comment:
do we need event_id here for `save_method_state` and `load_method_state`?  i feel like after your proposed change in handling event_id offset client side, we can probably remove it for the user facing api.